source/common/stream_info/stream_info_impl.h

```diff
@@ -375,6 +376,14 @@ struct StreamInfoImpl : public StreamInfo {
   void setIsShadow(bool is_shadow) { is_shadow_ = is_shadow; }
   bool isShadow() const override { return is_shadow_; }
 
+  void setDownstreamTransportFailureReason(absl::string_view failure_reason) override {
+    downstream_transport_failure_reason_ = std::string(failure_reason);
+  }
+
+  absl::string_view downstreamTransportFailureReason() const override {
+    return absl::string_view(downstream_transport_failure_reason_);
```

Comment:
Explicit `absl::string_view()`  conversion is redundant. The implicit conversion from string to string_view should be triggered

Reply:
My initial thinking is same,  but it will fail clang-asan check without this.

Reply:
Thanks, changed back. The mock make me think somewhere wired.